src/main/java/org/atlasapi/remotesite/bbc/nitro/GlycerinQueryExecutor.java

```
@@ -0,0 +1,89 @@
+        this.client = client;
+        this.currentIterator = queries.iterator();
+        this.iterator = new ProgrammeIterator(client, currentIterator);
+    }
+
+    @Override
+    public Iterator<Programme> iterator() {
+        return iterator;
+    }
+
+    private static class ProgrammeIterator implements Iterator<Programme> {
+
+        private Glycerin client;
+        private Iterator<ProgrammesQuery> queryIterator;
+        private GlycerinResponse<Programme> programmeGlycerinResponse;
+
+        private Iterator<Programme> currentProgramme;
+
+        public ProgrammeIterator(Glycerin client, Iterator<ProgrammesQuery> queryIterator) {
+            this.client = client;
+            this.queryIterator = queryIterator;
+        }
+
+        @Override
+        public boolean hasNext() {
+            if (currentProgramme != null && currentProgramme.hasNext()) {
+                return true;
+            }
+
+            if (queryIterator.hasNext()) {
+                try {
+                    this.programmeGlycerinResponse = client.execute(queryIterator.next());
+
+                    currentProgramme = programmeGlycerinResponse.getResults().iterator();
+                    if (currentProgramme.hasNext()) {
+                        return true;
+                    } else {
+                        if (queryIterator.hasNext()) {
+                            currentProgramme = client.execute(queryIterator.next()).getResults().iterator();
+                            if (currentProgramme.hasNext()) {
+                                return true;
+                            } else {
+                                return false;
+                            }
+                        } else {
+                            return false;
+                        }
+                    }
+                } catch (GlycerinException e) {
+                    log.warn("Couldn't get programmes from Nitro - {}", e);
```

This is **very** bad, you're swallowing an exception here. This should definitely propagate up rather than just pretending you've exhausted the iterator.